src/Microsoft.ML.TimeSeries/Deseasonality.cs

```diff
@@ -120,12 +122,10 @@ public void Deseasonality(ref double[] values, int period, ref double[] results)
                     results[i] = _stl.Residual[i];
                 }
             }
+            // invoke the back up deseasonality method if stl decompose fails.
             else
             {
-                for (int i = 0; i < values.Length; ++i)
-                {
-                    results[i] = values[i];
-                }
+                _backupFunc.Deseasonality(ref values, period, ref results);
             }
```

Comment:
(comment not specific to this line)

I don't see any tests associated with this PR. What is the behavior you are expecting to change from this PR? Is it possible to add some test for that?

Reply:
The behavior change of this PR is to bound the value of not anomaly point inside the lower / upper boundary. Checks are added to test this behavior.

---
In reply to: [480434195](https://github.com/dotnet/machinelearning/pull/5374#discussion_r480434195) [](ancestors = 480434195)